scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -233,6 +233,137 @@ object ContextAndArgs {
   }
 }
 
+trait ScioExecutor {
+  def run(sc: ScioContext): ScioExecutionContext
+}
+
+sealed trait ScioRunner extends ScioExecutor
```

Comment:
What's the reason for having `ScioRunner` and `ScioExecutor`? They can be the same trait/object?

Reply:
the idea is to allow the following:

```scala
val sc: ScioContext = ???
val testExecutor: ScioExecutor = ???
val runner = ScioRunner(testExecutor)

// case 1
sc.run(runner)

// case 2
runner.run(sc)
```

in either case inner context logic before `run` is preserved. User is only allowed to override the `ScioExecutor`